include/CppUTest/CppUTestConfig.h

```diff
@@ -211,13 +213,31 @@ typedef unsigned long long cpputest_ulonglong;
 /* Define some placeholders to disable the overloaded methods.
  * It's not required to have these match the size of the "real" type, but it's occasionally convenient.
  */
+
 #if defined(CPPUTEST_64BIT) && !defined(CPPUTEST_64BIT_32BIT_LONGS)
-typedef struct { char dummy[16]; } cpputest_longlong;
-typedef struct { char dummy[16]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 16
 #else
-typedef struct { char dummy[8]; } cpputest_longlong;
-typedef struct { char dummy[8]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 8
 #endif
+
+struct cpputest_longlong
+{
+#if defined(__cplusplus)
+  cpputest_longlong() {}
+  cpputest_longlong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
+struct cpputest_ulonglong
+{
+#if defined(__cplusplus)
+  cpputest_ulonglong() {}
+  cpputest_ulonglong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
```

Comment:
If you declare `struct cpputest_longlong`, in C++ you can refer to it just as `cpputest_longlong` but in C it must be referred to as `struct cpputest_longlong` unless there is also a typedef.

So yes, it's still possible to use it in C either way, but it's cleaner if the `struct` keyword is not required when using it, and this makes it more consistent with C++.  This is also important for unity with the case where `cpputest_longlong` is actually a typedef to the real type -- in this case obviously using `struct` is forbidden.  So it really does have to be a typedef.

As for working in C as-is, I've just realised that I only implemented long longs (and bools!) in the C mock API; I did not do so in the C test API, as I somehow missed that file entirely.  So it's working because they're not there. :sweat: 

Would it make things easier if I went back to the other PR and updated it to match this one, plus C-longlong support?